source/jormungandr/tests/kirin_realtime_tests.py

```diff
@@ -1278,6 +1279,136 @@ def test_no_rt_add_possible(self):
+        assert impacted_stops[0]['cause'] == 'on time'
+
+        assert bool(impacted_stops[1]['is_detour']) is True
+        assert impacted_stops[1]['cause'] == 'deleted for detour'
+        assert impacted_stops[1]['departure_status'] == 'unchanged'
+        assert impacted_stops[1]['arrival_status'] == 'deleted'
+
+        assert bool(impacted_stops[2]['is_detour']) is True
+        assert impacted_stops[2]['cause'] == 'added for detour'
+        assert impacted_stops[2]['departure_status'] == 'added'
+        assert impacted_stops[2]['arrival_status'] == 'added'
+
+
+@dataset(MAIN_ROUTING_TEST_SETTING)
+class TestKirinStopTimeOnDetourAndArrivesBeforeDeletedAtTheEnd(MockKirinDisruptionsFixture):
+    def test_stop_time_with_detour_and_arrival_before_deleted_at_the_end(self):
+        """
+        1. create a new_stop_time at C to replace existing one at A so that we have A deleted_for_detour
+        and C added_for_detour with arrival time < to arrival time of A (deleted)
+        2. Kraken rejects this disruption and is a BUG. Adjust this test after correction in kraken
+        """
+        disruptions_before = self.query_region('disruptions?_current_datetime=20120614T080000')
+        nb_disruptions_before = len(disruptions_before['disruptions'])
+
+        # New disruption with one stop_time same as base schedule, another one deleted and
+        # a new stop_time on stop_point:stopC added at the end
+        self.send_mock(
+            "vjA",
+            "20120614",
+            'modified',
+            [
+                UpdatedStopTime(
+                    "stop_point:stopB",
+                    arrival_delay=0,
+                    departure_delay=0,
+                    arrival=tstamp("20120614T080100"),
+                    departure=tstamp("20120614T080100"),
+                    message='on time',
+                ),
+                UpdatedStopTime(
+                    "stop_point:stopA",
+                    arrival_delay=0,
+                    departure_delay=0,
+                    arrival=tstamp("20120614T080102"),
+                    departure=tstamp("20120614T080102"),
+                    arrival_skipped=True,
+                    is_detour=True,
+                    message='deleted for detour',
+                ),
+                UpdatedStopTime(
+                    "stop_point:stopC",
+                    arrival_delay=0,
+                    departure_delay=0,
+                    arrival=tstamp("20120614T080101"),
+                    departure=tstamp("20120614T080101"),
+                    is_added=True,
+                    is_detour=True,
+                    message='added for detour',
+                ),
+            ],
+            disruption_id='stop_time_with_detour',
+        )
+
+        # Verify disruptions
+        disrupts = self.query_region('disruptions?_current_datetime=20120614T080000')
+        assert len(disrupts['disruptions']) == nb_disruptions_before
```

Comment:
We should at least comment it here to ease understanding when it changes, but here we enforce that the behavior is wrong :disappointed:
I propose a comment tag to use anywhere like that : `# TODO XFAIL` (and maybe a link to github issue if existing, coz JIRA is not nice for external people).
Did we track the task somewhere?